packit/upstream.py

```diff
@@ -68,34 +66,14 @@ def active_branch(self) -> str:
 
     def set_local_project(self):
         """ update self.local_project """
-        # TODO: ogr should have a method, something like this:
-        #       get_github_service(token, app_id, inst_id, cert_path) -> GithubService
-        #       the logic below should be the function
-        #       I want to leave this code here up the end of this sprint
         # TODO: in order to support any git forge here, ogr should also have a method like this:
         #       get_github_service_from_url(url, **kwargs):
         #       ogr should guess the forge based on the url; kwargs should be passed to the
         #       constructor in order to support the above
-        if (
-            self.config.github_app_id
-            and self.config.github_app_cert_path
-            and self.config.github_app_installation_id
-        ):
-            logger.info("Authenticating with Github using a Githab app.")
-            private_key = Path(self.config.github_app_cert_path).read_text()
-            integration = github.GithubIntegration(
-                self.config.github_app_id, private_key
-            )
-            token = integration.get_access_token(self.config.github_app_installation_id)
-            gh_service = GithubService(token=token, read_only=self.config.dry_run)
-        else:
-            logger.debug("Authenticating with Github using a token.")
-            gh_service = GithubService(
-                token=self.config.github_token, read_only=self.config.dry_run
-            )
+        if not self.local_project.git_service:
+            self.local_project.git_service = get_github_service(self.config)
+            self.local_project.refresh_the_arguments()  # get git project from newly set git service
```

Comment:
:+1:

Reply:
that's Rado's code, I just updated it a bit